src/Agent.Worker/TestResults/FileZipper.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    [ServiceLocator(Default = typeof(FileZipper))]
+    public interface IZipper : IAgentService
+    {
+        void CreateZipFile(string zipfileName, IEnumerable<string> files);
```

Comment:
I think you go one of two ways - private function (not much code here) or move it to common util and create an archive class